### ISSUE TITLE
Question about the tables dotted keys define.

### ISSUE THREAD

**Issue by Bagolly** [toml-lang/toml#1032]
In the [Table](https://toml.io/en/v1.0.0#table) section of the v1.0.0 specification, it  says:

>Dotted keys create and define a table for each key part before the last one, provided that such tables were not previously created.

and that

>Naming rules for tables are the same as for keys (see definition of [Keys](https://toml.io/en/v1.0.0#keys) above).

Since the rules are the same, this implied to me that there is no difference between tables created via
- dotted keys in _table headers_  or
- dotted keys in _key/value pairs_

and this made perfect sense. However (right after the first quote), it then says that

> Since tables cannot be defined more than once, redefining such tables using a [table] header is not allowed.

Which, to me, seemed to contradict

> You don't need to specify all the super-tables if you don't want to. TOML knows how to do it for you.

because while the former rule would disallow this:
```toml
[x.y.z.w]
key = "some value"
[x]
key = "some value"
```
as well as this:
```toml
 x.y.z.w.key = "some value"
 [x]
 key = "some value"
```
_(tested with an approved C# implementation called Tomlyn.)_

Since dotted keys _create and define_ a table, and _redefining such tables using a [table] header is not allowed_, I would expect attempting to redefine `x` to result in an error in both cases.

But the first one is valid, while the second one is not, despite `x` being the same "implicitly-created via dotted key" super-table in both cases. 

The only difference I could find between the two, is that the implicit table `x`  was defined in a table header in the first one, and a key/value pair's key in the second one, but this should not matter; both are just "dotted keys".

With all that said, I would appreciate some help with the following:

1. Is there an implied difference I'm missing between the tables dotted keys define in table headers and key/value pairs? Is this related to #846?
3. In the spec, what is the difference between **defining**, **creating**, and most importantly, **specifying** tables?

**Comment by eksortso** [comment]
> I suppose the confusion for me came from the spec saying dotted tables create and **define** their tables, but this isn't actually true for dotted keys in table headers.

To be clear, we have never referred to the names in table headers to be "dotted keys." Those are only used in key/value pairs. Historically, table names are absolute paths through table (and array-of-table) trees, so we didn't call them keys, though they use the same namespaces, so to speak.

**Comment by Bagolly** [comment]
> To be clear, we have never referred to the names in table headers as "dotted keys."

That is true, I'm not denying my oversight; however, the spec never clarifies that they _are_ different either, and both the written and ABNF spec uses the keys for key/value pairs (and thereby dotted keys) to define what is valid within `[]`.
ABNF:
```abnf
key = simple-key / dotted-key
;(...)
std-table = std-table-open key std-table-close
```
Written:
1. "Naming rules for tables are the same as for keys", linking to **Keys**. 
2. From **Keys**, scrolling down to **Dotted keys**, it said "For details regarding the tables that dotted keys define, refer to the [Table](https://toml.io/en/v1.0.0#table) section below.", linking back to Tables.

Because of this, and since "dotted table(s)" or a similar term is never used in the spec, I assumed  that a 'dotted sequence' (for lack of better term) in a _header_ defines the same table as a dotted sequence in a _key/value pair_, because they share the same definition (in both text and ABNF), and there is also no other name given to a table header of the form `[a.b]` that I could find.

I suppose some explicit clarification, for example "_While the syntax for headers and dotted keys are the same, the tables they define are not equivalent_", (and then explain the difference) after the section linking back to Keys would've made things more clear.

But then again if I'm the only one who misunderstood this part of the spec, then the issue lies with me, and since
I've already fixed the bugs related to this (using the clarification I received here), it didn't really matter.
